algorithms/isomorphism_problem.py

```diff
@@ -13,8 +14,10 @@ def are_isomorph(G, H):
     :param G, H: The two graphs of which it must be determined if there is an isomorphism.
     :return: Boolean that indicates if graph G and H are isomorph or not.
     """
-    G_disjoint_union = G + H
+    if not could_be_isomorphic(G, H):
+        return False
 
+    G_disjoint_union = G + H
```

Comment:
De disjoint union stap wordt ook al gedaan in could_be_isomorphic(G, H)

Reply:
Het resultaat wordt alleen niet gereturned.

Reply:
Dat klopt, omdat ik met de methode could_be_isomorphic als enige effect wilde hebben dat het het resultaat teruggeeft of het wel of niet isomorph kan zijn, dus ik wilde niet binnen deze functie de disjoint union maken en teruggeven.

Ik snap inderdaad dat het nu dubbel is, maar ga er nog even over denken of dat dan een oke effect is om dan de disjoint union terug te geven als je verder moet gaan met ismorphismen zoeken.

Reply:
Ik denk dat ik het zo laat, want het bepalen van de disjoint union heeft vooral te maken met het feit dan ons algoritme op het moment zo werkt. Maar als we in de toekomst een algoritme zouden hebben waarin je de grafen apart moet geven als argumenten heeft het geen zin om de disjoint union te maken.

Reply:
In de huidige structuur past het returnen inderdaad niet goed.
Als hierdoor echt significante snelheids winst mogelijk is kan het wat mij betreft later worden opgelost.